dts/riscv/wch/ch32v00x.dtsi

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) 2024 Michael Hope
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <mem.h>
+#include <zephyr/dt-bindings/gpio/gpio.h>
+#include <zephyr/dt-bindings/i2c/i2c.h>
+#include <zephyr/dt-bindings/clock/ch32v00x-clocks.h>
+
+/ {
+	#address-cells = <1>;
+	#size-cells = <1>;
+
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu0: cpu@0 {
+			device_type = "cpu";
+			compatible = "wch,qingke-v2";
+			reg = <0>;
+		};
+	};
+
+	soc {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "simple-bus";
+		ranges;
+
+		debug0: debug@e00000f0 {
+			compatible = "wch,qingkev2-debug";
+			reg = <0xe00000f0 0x40>;
+		};
+
+		sram0: memory@20000000 {
+			compatible = "mmio-sram";
+			reg = <0x20000000 0x800>;
+		};
+
+		flash: flash-controller@40022000 {
+			compatible = "wch,ch32v00x-flash-controller";
+			reg = <0x40022000 0x400>;
+
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			flash0: flash@0 {
+				compatible = "soc-nv-flash";
+				/* TODO: actually 8000000 */
+				reg = <0 0x4000>;
+			};
+		};
+
+		pfic: interrupt-controller@e000e000 {
+			compatible = "wch,pfic";
+			#address-cells = <0>;
+			#interrupt-cells = <1>;
+			interrupt-controller;
+			reg = <0xe000e000 16>;
+			status = "okay";
+		};
+
+		systick: systimer@e000f000 {
+			compatible = "wch,systick";
+			reg = <0xe000f000 16>;
+			status = "okay";
+			interrupt-parent = <&pfic>;
+			interrupts = <12>;
+		};
+
+		tim2: counter@40000000 {
+			compatible = "wch,gptm";
+			reg = <0x40000000 16>;
+			pwm-controller;
+			#pwm-cells = <3>;
+			prescaler = <1>;
+			clocks = <&rcc CH32V00X_CLOCK_TIM2>;
+		};
```

Comment:
You should drop all entries that do not have a driver. It is easy to review and catch errors when a new driver is introduced. It is easy to maintainer to check dts bindings, devicetree entries, the driver it self, the board entries and the tests.

Reply:
Done now - thanks!